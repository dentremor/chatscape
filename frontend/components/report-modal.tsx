import {
  Modal,
  ModalVariant,
  Button,
  Form,
  FormGroup,
  TextArea,
} from "@patternfly/react-core";
import { useState } from "react";

export const ReportModal = ({
  name,
  modalOpen,
  closeModal,
  reportContact,
}: {
  name: string;
  modalOpen: boolean;
  closeModal: () => void;
  reportContact: (comment: string) => void;
}) => {
  const [reportFormComment, setReportFormComment] = useState("");

  return (
    <Modal
      bodyAriaLabel="Report modal"
      variant={ModalVariant.small}
      title={`Report ${name}`}
      isOpen={modalOpen}
      onClose={closeModal}
      actions={[
        <Button key="report" variant="danger" type="submit" form="report-form">
          Report
        </Button>,
        <Button key="cancel" variant="link" onClick={() => closeModal}>
          Cancel
        </Button>,
      ]}
    >
      <Form
        id="report-form"
        onSubmit={() => {
          reportContact(reportFormComment);
          closeModal();
        }}
        noValidate={false}
      >
        <FormGroup
          label="Your comment and the messages in violations of our policy"
          isRequired
          fieldId="report-form-comment"
        >
          <TextArea
            isRequired
            required
            id="report-form-comment"
            name="report-form-comment"
            value={reportFormComment}
            onChange={(v) => setReportFormComment(v)}
          />
        </FormGroup>
      </Form>
    </Modal>
  );
};
