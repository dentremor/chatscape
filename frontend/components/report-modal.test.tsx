import { render } from "@testing-library/react";
import { ReportModal } from "./report-modal";

describe("ReportModal", () => {
  it("renders with modal open", () => {
    const mockCloseModal = jest.fn();
    const mockReportContact = jest.fn();

    expect(
      render(
        <ReportModal
          name="Contact Name"
          modalOpen={true}
          closeModal={mockCloseModal}
          reportContact={mockReportContact}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders with modal closed", () => {
    const mockCloseModal = jest.fn();
    const mockReportContact = jest.fn();

    expect(
      render(
        <ReportModal
          name="Contact Name"
          modalOpen={false}
          closeModal={mockCloseModal}
          reportContact={mockReportContact}
        />
      )
    ).toMatchSnapshot();
  });
});
