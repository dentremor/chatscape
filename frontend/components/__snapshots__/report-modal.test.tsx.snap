// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportModal renders with modal closed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class=""
  >
    <div />
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ReportModal renders with modal open 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="pf-c-backdrop__open"
  >
    <div
      aria-hidden="true"
    />
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-labelledby="pf-modal-part-1"
            aria-modal="true"
            class="pf-c-modal-box pf-m-sm"
            data-ouia-component-id="OUIA-Generated-Modal-small-1"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
          >
            <button
              aria-disabled="false"
              aria-label="Close"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
            <header
              class="pf-c-modal-box__header"
            >
              <h1
                class="pf-c-modal-box__title"
                id="pf-modal-part-1"
              >
                <span
                  class="pf-c-modal-box__title-text"
                >
                  Report Contact Name
                </span>
              </h1>
            </header>
            <div
              aria-label="Report modal"
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
              role="region"
            >
              <form
                class="pf-c-form"
                id="report-form"
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="report-form-comment"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Your comment and the messages in violations of our policy
                      </span>
                      <span
                        aria-hidden="true"
                        class="pf-c-form__label-required"
                      >
                         
                        *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <textarea
                      aria-invalid="false"
                      class="pf-c-form-control"
                      id="report-form-comment"
                      name="report-form-comment"
                      required=""
                    />
                  </div>
                </div>
              </form>
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-danger"
                data-ouia-component-id="OUIA-Generated-Button-danger-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                form="report-form"
                type="submit"
              >
                Report
              </button>
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-link"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Cancel
              </button>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
